### ISSUE TITLE
Console input not working for `iex --remsh` in Elixir 1.15.2

### ISSUE THREAD

**Issue by lukyanov** [elixir-lang/elixir#12794]
### Elixir and Erlang/OTP versions

Erlang/OTP 25 [erts-13.2.2.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Elixir 1.15.2 (compiled with Erlang/OTP 24)

### Operating system

Mac OS 13.4 (Alpine in Docker has the same issue though)

### Current behavior

1. `mix release`
2. `_build/dev/rel/core/bin/core start`
3. `_build/dev/rel/core/bin/core remote` (in another shell)

The remote node is connected to the release's one, but the console input is blocked (can't type anything). There is no promt as well:
```
_build/dev/rel/core/bin/core remote
Erlang/OTP 25 [erts-13.2.2.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]


```

Erlang and Elixir are isntalled with `asdf`.

### Expected behavior

The same steps works as expected in Elixir 1.14.5:
```
$ _build/dev/rel/core/bin/core remote
Erlang/OTP 25 [erts-13.2.2.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Interactive Elixir (1.14.5) - press Ctrl+C to exit (type h() ENTER for help)
iex(core@127.0.0.1)1>

**Comment by lukyanov** [comment]
I tried `iex` directly with the same result:
```
$ iex --name remote@127.0.0.1 --cookie 123 --remsh "core@127.0.0.1"
Erlang/OTP 25 [erts-13.2.2.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]


```

**Comment by lukyanov** [comment]
Trired with `Erlang 24.3.4.13`, still no console input.

**Comment by josevalim** [comment]
Fixed in main/v1.15 branches.

**Comment by josevalim** [comment]
It will be out with v1.15.3.
